third_party/move/move-compiler/src/cfgir/optimize/inline_blocks.rs

```diff
@@ -88,15 +93,35 @@ fn inline_single_target_blocks(
             // be the target of at least 2 jumps: the jump to the loop and the "continue" jump
             // This is always true as long as we start the count for the start label at 1
             sp!(_, Command_::Jump { target, .. }) if single_jump_targets.contains(target) => {
+                // Note that only the last merged block will be left for cur.
                 remapping.insert(cur, *target);
-                let target_block = working_blocks.remove(target).unwrap();
-                block.pop_back();
-                block.extend(target_block);
+                match working_blocks.remove(target) {
```

Comment:
Consider rewriting (for reduced code duplication) to:
```
                let target_block = working_blocks.remove(target).unwrap_or_else(|| {
                    finished_blocks.remove(target).unwrap_or_else(|| {
                        panic!(
                            "ICE: Target {} not found in working_blocks or finished_blocks",
                            target
                        )
                    })
                });
                block.pop_back();
                block.extend(target_block);
```

Reply:
done.